### DISCUSSION TITLE
[2.0.0b3] Incompatible types in assignment using MyPy

### DISCUSSION THREAD

**Discussion by graille** [sqlalchemy/sqlalchemy#8845] [Usage Questions]
I'm currently moving my SQLAlchemy codebase to 2.0. So I just installed 2.0.0b3
I'm also trying to be mypy compliant. I read the doc about 2.0 and type hinting, but when I'm running mypy on my code, I have the following error :

> error: Cannot assign to a type  [misc] (This one is on the line Base = declarative_base(cls=Base))
> error: Incompatible types in assignment (expression has type "Column[<nothing>]", variable has type "UUID")  [assignment]
> error: Incompatible types in assignment (expression has type "Column[str]", variable has type "str")  [assignment]
> error: Incompatible types in assignment (expression has type "Column[str]", variable has type "str")  [assignment]

This is my entity declaration using SQLAlchemy ORM:

```python
import uuid

from sqlalchemy import Column, String, Text
from sqlalchemy.orm import declarative_base
from sqlalchemy_utils import UUIDType


class Base:
    # Source: https://docs.sqlalchemy.org/en/14/changelog/migration_20.html#migration-to-2-0-step-six-add-allow-unmapped-to-explicitly-typed-orm-models
    __allow_unmapped__ = True


Base = declarative_base(cls=Base)


class Community(Base):
    __tablename__ = 'communities'

    id: uuid.UUID = Column(UUIDType, primary_key=True, default=uuid.uuid4)
    name: str = Column(String(255), nullable=False)
    description: str = Column(Text, nullable=False, default="")
```

I don't have any mypy configuration enabled. And I just did a fresh installation of my local Python environnent.

Any Idea ?

**Comment by CaselIT** [comment]
Hi,

Well those errors are correct. If you want to correctly type check the models you will need to use the new syntax for v2.
Otherwise the mypy plugin is required (I think it will support these cases)

**Reply by graille** [reply]
Well, according to https://docs.sqlalchemy.org/en/14/changelog/migration_20.html#migration-to-2-0-step-six-add-allow-unmapped-to-explicitly-typed-orm-models adding `__allow_unmapped__ = True` should fix these kind of error

**Reply by zzzeek** [reply]
are you using the mypy plugin?  that would be required for the above model to type correctly.   `__allow_unmapped__` refers to runtime behavior only.

**Reply by zzzeek** [reply]
the doc you refer towards above is to allow a 1.4 application to run under 2.0 with an absolute minimum of changes.  To migrate your models to 2.0 style fully, that's in the 2.0 documentation, starting at https://docs.sqlalchemy.org/en/20/changelog/whatsnew_20.html#migrating-an-existing-mapping

**Reply by zzzeek** [reply]
ive added a note to the document for 1.4 and 2.0
